packages/neo4j-driver/test/internal/transaction-executor.test.js

```diff
@@ -320,178 +315,177 @@ describe('#unit TransactionExecutor', () => {
+      executor.pipelineBegin = pipelineBegin
+      const usedTransactions = []
+      const realWork = throwingTransactionWork(errorCodes, 42)
+
+      const result = await executor.execute(transactionCreator(), async tx => {
+        expect(tx).toBeDefined()
+        usedTransactions.push(tx)
+        if (pipelineBegin) {
+          await tx
+        }
+        return realWork()
+      })
+
+      // work should have failed 'failures.length' times and succeeded 1 time
+      expect(usedTransactions.length).toEqual(errorCodes.length + 1)
+      expectAllTransactionsToBeClosed(usedTransactions)
+      expect(result).toEqual(42)
+      verifyRetryDelays(fakeSetTimeout, errorCodes.length)
     }
 
     async function testRetryWhenTransactionWorkThrowsAndRollbackFails (
       txWorkErrorCodes,
       rollbackErrorCodes
     ) {
-      const fakeSetTimeout = setTimeoutMock.install()
-      try {
-        const executor = new TransactionExecutor()
-        executor.pipelineBegin = pipelineBegin
-        const usedTransactions = []
-        const realWork = throwingTransactionWork(txWorkErrorCodes, 424242)
-
-        const result = await executor.execute(
-          transactionCreator([], rollbackErrorCodes),
-          tx => {
-            expect(tx).toBeDefined()
-            usedTransactions.push(tx)
-            return realWork()
-          }
-        )
-
-        // work should have failed 'failures.length' times and succeeded 1 time
-        expect(usedTransactions.length).toEqual(txWorkErrorCodes.length + 1)
-        expectAllTransactionsToBeClosed(usedTransactions)
-        expect(result).toEqual(424242)
-        verifyRetryDelays(fakeSetTimeout, txWorkErrorCodes.length)
-      } finally {
-        fakeSetTimeout.uninstall()
-      }
+      const { fakeSetTimeout, executor } = createTransactionExecutorWithFakeTimeout()
+      executor.pipelineBegin = pipelineBegin
+      const usedTransactions = []
+      const realWork = throwingTransactionWork(txWorkErrorCodes, 424242)
+
+      const result = await executor.execute(
+        transactionCreator([], rollbackErrorCodes),
+        tx => {
+          expect(tx).toBeDefined()
+          usedTransactions.push(tx)
+          return realWork()
+        }
+      )
+
+      // work should have failed 'failures.length' times and succeeded 1 time
+      expect(usedTransactions.length).toEqual(txWorkErrorCodes.length + 1)
+      expectAllTransactionsToBeClosed(usedTransactions)
+      expect(result).toEqual(424242)
+      verifyRetryDelays(fakeSetTimeout, txWorkErrorCodes.length)
     }
   })
 })
 
+function createTransactionExecutorWithFakeTimeout (...args) {
+  const fakeSetTimeout = new TimeoutsMock()
+  const dependencies = {
+    setTimeout: fakeSetTimeout.setTimeout,
+    clearTimeout: fakeSetTimeout.clearTimeout
+  }
+
+  // args should have at least 5 elements for place the dependencies
+  if (args.length < 5) {
+    args.length = 5
```

Comment:
TIL: in JS you can just assign a length to an array  :exploding_head:

But please help me understand what it changes (other than the length attribute). `args[4]` will be `undefined` either way, won't it?

And apparently (tested with node), assigning elements out of bounds is no issue
```javascript
> l = [1, 2]
[ 1, 2 ]
> l[4] = 1
1
> l
[ 1, 2, <2 empty items>, 1 ]

```

Reply:
The other arguments will be undefined. I didn't know I can assign elements out of bound. So, I will do it instead. The code will looks cleaner.